### ISSUE TITLE
Removing old images

### ISSUE THREAD

**Issue by netbrain** [netbrain/zwift#156]
Consider removing old images (1 year retention?) so that we don't accumulate images with hub.docker.com

See https://stackoverflow.com/questions/44209644/how-do-i-delete-a-docker-image-from-docker-hub-via-command-line for a possible solution.

**Comment by relief-melone** [comment]
how about something like this (podman just for illustration. should work with containerTool just as well)
```sh
IMAGES_TO_DELETE=$(podman images --filter reference!="<current-image>:<current-version>" --filter <current-image> --filter until=<max-age-in-hours>h --format {{.ID}})
podman rmi $IMAGES_TO_DELETE
```
This basically gives you all images that are the current image but not the current tag that have an age of or older than max age and print out just the ids of those images.
After that delete those images. Of course this could just as well be a oneliner

**Comment by netbrain** [comment]
Yes this is important too. But this issue is regarding housekeeping on dockerhub, where I serve the Zwift container from.

**Comment by glennvl** [comment]
There's an official docker hub api that supports deleting images now.

<https://docs.docker.com/reference/api/registry/latest/#tag/delete>

Could be combined with <https://docs.docker.com/reference/api/hub/latest/#tag/repositories/operation/ListRepositoryTags> to get all tags to delete all except the last 3 (or however many you want to keep).

**Comment by netbrain** [comment]
For now, I'm just doing it manually whenever I feel like it. But yeah, at some point this should be automated.

Thanks for looking into it.
